packages/cdk8s-plus/src/pod.ts

```diff
@@ -82,7 +82,7 @@ export class PodSpec implements IPodSpec {
     this.restartPolicy = props.restartPolicy;
     this.serviceAccount = props.serviceAccount;
 
-    this._containers = props.containers ?? [];
+    this._containers = props.containers?.map(c => new Container(c)) ?? [];
```

Comment:
Worth calling `addContainer()` instead of reimplementing it so mutating the internal data structure happens in one place.